pallets/author-mapping/src/lib.rs

```diff
@@ -236,10 +236,10 @@ pub mod pallet {
 	#[pallet::storage]
 	#[pallet::getter(fn account_and_deposit_of)]
```

Comment:
```suggestion
	#[pallet::getter(fn mapping)]
```
convention is for the getter to use the same name as the map, not sure when this changed or if you want to keep this change

Reply:
That's a good point. At first I wanted to leave it unchanged so I wouldn't have to change everything that called it. But it turns out it is only called from within the pallet itself, so I did change it.

And actually, changing the name of the storage item is also bad. This is going to break Subscan, and possibly Polkadot JS Apps. I wish I knew a way to keep the same old storage name but change the schema.

Reply:
I wonder if this migration would work if I wrote it like this:
* Read all the mappings into memory
* Prune the old storage
* Write all mappings from memory into the new storage.

That would allow us to not change the storage name (or prefix) and also not worry about mutating the storage while iterating over it. Obviously the potential problem is if we run out of memory. I think in the handful of currently-live moonbeam networks, that is not likely to be a problem.

Reply:
There are utils for doing basically what you described: https://github.com/paritytech/substrate/blob/master/frame/support/src/migrations.rs

You can also find various examples of migrations by looking for directories named `migrations` or `migrations.rs` files in Substrate.